commons-vfs2/src/main/java/org/apache/commons/vfs2/provider/sftp/SftpFileSystem.java

```diff
@@ -296,4 +309,17 @@ private int executeCommand(final String command, final StringBuilder output) thr
         channel.disconnect();
         return channel.getExitStatus();
     }
+
+    /**
+     * Some SFTP-only servers disable the exec channel.
+     * 
+     * Attempt to detect this by calling getUid.
+     */
+    private void detectExecDisabled() {
+        try {
+            getUId();
+        } catch(JSchException | IOException e) {
+            execDisabled = true;
```

Comment:
Shouldn't the exception be stored here so that it can be queries by applications or tests? It would be a pain to have to step into the debugger to figure out what the real failure is here.

Reply:
Might be useful. I'm not familiar with the way VFS does error handling/logging. However, an exception here will either indicate the absence of the exec channel (which isn't really a problem and which is what I'm trying to detect), or something serious. In the case of "something serious", the exception (or a similar one) will surface when the SftpFileSystem instance is used.

Reply:
Log on debug such that we can always turn on debug logging for VFS to get exceptions.
( log.debug(....); )